### PR TITLE
 Extension Manager: Sort plugins/templates by name #1018 

### PR DESCRIPTION
**Pull request by DrPanic**
Extension Manager: Sort plugins/templates by name #1018 
Sort for Templates is added too

### CODE REVIEW

File: lib/plugins/extension/helper/list.php [hunk 1]
```diff
@@ -562,5 +562,24 @@ function make_status(helper_plugin_extension_extension $extension) {
...
+     *
+     * @param array $array The Plugin-Array
+     * @return array The sorted Plugin-Array
+     */
+	 function sortDisplayName(array $array) {
+		$extension = $this->loadHelper('extension_extension');
+		$sortedArray = array();
+		
+	    foreach($array as $name) {
+            $extension->setExtension($name);
```
Comment (splitbrain) [inline lib/plugins/extension/helper/list.php]: the indention is messed up here

**Comment by splitbrain** [comment]
Your commit messages are not very useful. Did you edit via the github interface? A proper fork/checkout/edit/test/commit/pr workflow would be preferable here.

I'm unsure about the way the sorting happens here. Iterating and setting up an extension object happens multiple times now. I wonder if it would be better to create a basename=>displayname plugin list in the first place and then simply sort by value.

**Comment by Tim222** [comment]
Sorry, werde demnächst mehr Informationen dazupacken. Die hässlichen Einrückungen kommen davon, dass ich zur Zeit über Notepad++ editiere, lokal teste und die Änderung dann in das Webpanel kopiere. Werde einen durchgängigen Workflow anstreben.

Zum Sortieren:
Du würdest bevorzugen, dass die Funktion die Plugins direkt als basename =>displayname einliest und dann entsprechend sortiert? Oder die getList() dazu abändern? Wobei ich für den Displaynamen ja dann trotzdem entweder direkt die Infodatei parsen müsste oder das Plugin als Object laden müsste.

**Comment by splitbrain** [comment]
yeah, I'm not sure what the best way would be. I just noticed that there's basically the same thing happening multiple times.

**Comment by selfthinker** [comment]
There are currently tabs and trailing whitespace in the code. We use 4 spaces instead of tabs. (I guess that's what the scrutinizer inspection would tell us, but I cannot check that as their website seems to be down right now.)

**Comment by micgro42** [comment]
Sorting in the extension manager was implemented in #2893

**Pull request closed without merging**